src/awkward/_v2/contents/regulararray.py

```diff
@@ -1000,70 +1000,198 @@ def _reduce_next(
         keepdims,
         behavior,
     ):
-        out = self.toListOffsetArray64(True)._reduce_next(
-            reducer,
-            negaxis,
-            starts,
-            shifts,
-            parents,
-            outlength,
-            mask,
-            keepdims,
-            behavior,
-        )
 
-        if not self._content.dimension_optiontype:
-            branch, depth = self.branch_depth
-            convert_shallow = negaxis == depth
-            convert_deep = negaxis + 2 == depth
+        branch, depth = self.branch_depth
+        nextlen = len(self) * self._size
+
+        if negaxis == depth and not branch:
```

Comment:
Same as https://github.com/scikit-hep/awkward/pull/1811/files#r1000929199

```suggestion
        if not branch and negaxis == depth:
```